compat/src/forwardRef.js

```diff
@@ -0,0 +1,31 @@
+import { options } from 'preact';
+import { assign } from './util';
+
+let oldVNodeHook = options.vnode;
+options.vnode = vnode => {
```

Comment:
wondering if we should maybe keep these "side-effects" in a central place, maybe entrypoint so they don't bite us at some point

Reply:
In my next PR I was actually playing with ways to only install options when a consumer needs them, to try and make `preact/compat` more tree-shakeable. This works well if each module contains its own option hook so it's easy to know what code is needed for that feature to work.

Though it does have a downside of potentially leading to duplication of options, but I think if we can come up with a pattern that enables tree-shaking, it might be worth it.

Reply:
Hmm, maybe putting them in the same files should be in that PR then, if you can figure out a way to do it 🤷‍♀️